tests/objects/test_world_object.py

```diff
@@ -390,3 +391,19 @@ def test_bounding_box():
     )
     point_with_children.add(scene)
     assert point_with_children.get_bounding_box().tolist() == [[-5, 0, 0], [9, 3, 6]]
+
+
+def test_scale_preservation():
+    # Add a point that is transformed, to make sure that is taken into account
+    ob = gfx.WorldObject()
+    s = (1, -2, 3)
+    ob.local.scale = s
+    # without scale preservation in matrix compose -> decompose roundtrip
+    # ob.local.scale becomes (-1, 2, 3)
+    npt.assert_array_equal(ob.local.scale, s)
+
+    child = gfx.WorldObject()
+    ob.add(child)
+    npt.assert_array_equal(child.local.scale, [1, 1, 1])
+    # we don't preserve the WORLD scale component
+    npt.assert_array_equal(child.world.scale, [-1, 2, 3])
```

Comment:
What does this mean in practice? That it's fine to flip the scale factors of e.g. the camera (which is the most common scenario, I think), but not of objects that (may) have children?

Reply:
It means that we can guarantee scaling will be preserved if you set it on local.scale, but not if you set it on world.scale.

Reply:
But this test also seems to show that negative scale factors don't transfer (correctly) to their children, right?

Reply:
Well, they're equivalent, so I wouldn't say they're incorrect.

Reply:
Right, and that's ok, because in this case there is no roundtrip.

Reply:
Exactly!